### ISSUE TITLE
MPNet: Masked and Permuted Pre-training for Language Understanding

### ISSUE THREAD

**Issue by RyanHuangNLP** [huggingface/transformers#4308]
# 🌟 New model addition

## Model description

[MPNet: Masked and Permuted Pre-training for Language Understanding](https://arxiv.org/pdf/2004.09297.pdf)
use both Masked Language Model and Permuted Language Model

## Open source status

* [MPNet ] the model implementation is available: (https://github.com/microsoft/MPNet)
* [MPNet pretrain weight ] the model weights are available: (https://modelrelease.blob.core.windows.net/pre-training/MPNet/mpnet.example.pt)
* [@tobyoup@StillKeepTry ] who are the authors: (mention them, if possible by @gh-username)

**Comment by tobyoup** [comment]
Hi huggingface/transformers,

I am Xu Tan from Microsoft Research. Thanks for that you mentioned MPNet in huggingface/transformers repo.

MPNet is an advanced pre-trained models for language understanding, which inherits the advantages of BERT and XLNet, while well address their problems.  We have MPNet pre-trained models, codes (the version for huggingface/transformers) and downstream task results ready.

How can we proceed to incorporate MPNet into huggingface/transformers repo? Thanks very much!


Best,
Xu

**Comment by tobyoup** [comment]
+ RyanHuangNLP

From: Xu Tan
Sent: Sunday, July 5, 2020 6:58 PM
To: huggingface/transformers <reply@reply.github.com>; huggingface/transformers <transformers@noreply.github.com>
Subject: RE: [huggingface/transformers] MPNet: Masked and Permuted Pre-training for Language Understanding (#4308)

Hi huggingface/transformers,

I am Xu Tan from Microsoft Research. Thanks for that you mentioned MPNet in huggingface/transformers repo.

MPNet is an advanced pre-trained models for language understanding, which inherits the advantages of BERT and XLNet, while well address their problems.  We have MPNet pre-trained models, codes (the version for huggingface/transformers) and downstream task results ready.

How can we proceed to incorporate MPNet into huggingface/transformers repo? Thanks very much!


Best,
Xu

**Comment by LysandreJik** [comment]
Hi @tobyoup, it's very cool that you want to contribute your architecture!

You can follow this [checklist](https://github.com/huggingface/transformers/tree/master/templates/adding_a_new_model) to add your model. Once you open a PR, we'll be very happy to help you finish the integration!

**Comment by RyanHuangNLP** [comment]
@tobyoup @LysandreJik Is there any update about MPNET in transformers?

**Comment by tobyoup** [comment]
We are actively preparing the integration codes, and will be ready these weeks.

**Comment by trouble-maker007** [comment]
@tobyoup Is MPNET. ready in transformers?

**Comment by trouble-maker007** [comment]
@tobyoup @LysandreJik Is there any update about MPNET in transformers?

**Comment by LysandreJik** [comment]
The PR for the MPNET integration has not seen any progress yet. @tobyoup, don't hesitate to ping us if you need help or pointers towards the integration.

**Comment by tobyoup** [comment]
Thanks for the reminder! We are adapting the codes these days and we expect to give the first PR by this weekend.

Thanks
Xu
